jerry-core/include/jerryscript-core.h

```diff
@@ -159,6 +159,18 @@ typedef struct
 } jerry_property_descriptor_t;
 
 /**
+ * Description of JerryScript heap memory stats.
+ * It is for memory profiling.
+ */
+typedef struct
+{
+  size_t version; /**< the version of the stats struct */
```

Comment:
I'm not really convinced `version` is needed.
IMO, the struct and field types and names are a semantic API contract, but IMO the struct layout isn't an ABI (the exact layout is target- & compiler dependent after all...)
I think it's safe to assume that the users of JerryScript have control over what version of Jerry they use.
If the semantics change (i.e. field is renamed, types are changed, ...) the compiler will tell you.